### ISSUE TITLE
🐛 [cloud_functions] Added to flutter project and breaks web app

### ISSUE THREAD

**Issue by hparicha** [firebase/flutterfire#9504]
## Bug report
Added to pubspec.yaml `cloud_functions: ^3.3.7`
Have the emulator running (helloWorld)
Added to main.dart before runApp `FirebaseFunctions.instance.useFunctionsEmulator('localhost', 5001);`

**Describe the bug**
When run the app debug in Chrome, get the red screen and
`The following JSNoSuchMethodError was thrown building Builder(dirty):
TypeError: Cannot read properties of undefined (reading 'auth')
The relevant error-causing widget was:
  MaterialApp
`

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.18.0-271.4.beta
Flutter SDK 3.3.0-0.3.pre
sign_in_firebase_auth 1.0.0+1

dependencies:
- beamer 1.3.0 [flutter flutter_web_plugins]
- build_runner 2.1.7 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- cached_network_image 3.2.1 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- charts_flutter 0.12.0 [charts_common collection flutter intl logging meta]
- cloud_firestore 3.1.8 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cloud_functions 3.3.7 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- collection 1.16.0
- confetti 0.7.0 [flutter vector_math]
- cupertino_icons 1.0.4
- dartx 1.1.0 [characters collection crypto meta path time]
- equatable 2.0.3 [collection meta]
- firebase_auth 3.3.7 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.12.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_storage 10.2.7 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_html 2.2.1 [html csslib flutter_layout_grid video_player chewie webview_flutter chewie_audio flutter_svg flutter_math_fork collection numerus flutter]
- flutter_login_facebook 1.2.1 [flutter list_ext]
- flutter_riverpod 1.0.3 [collection flutter meta riverpod state_notifier]
- font_awesome_flutter 9.2.0 [flutter]
- freezed 1.1.1 [analyzer build build_config collection meta source_gen freezed_annotation]
- freezed_annotation 1.1.0 [collection json_annotation meta]
- get 4.6.1 [flutter]
- google_fonts 3.0.1 [flutter http path_provider crypto]
- google_sign_in 5.2.3 [flutter google_sign_in_platform_interface google_sign_in_web]
- hive 2.2.3 [meta crypto]
- hive_flutter 1.1.0 [flutter hive path_provider path]
- http 0.13.4 [async http_parser meta path]
- intl 0.17.0 [clock path]
- json_annotation 4.4.0 [meta]
- jwt_decode 0.3.1
- logger 1.1.0
- pdf 3.7.1 [archive barcode crypto image meta path_parsing vector_math xml]
- percent_indicator 4.0.0 [flutter]
- photo_view 0.13.0 [flutter]
- printing 5.7.2 [ffi flutter flutter_web_plugins http image js meta pdf plugin_platform_interface]
- provider 6.0.2 [collection flutter nested]
- rxdart 0.27.3
- url_launcher 6.0.20 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- uuid 3.0.5 [crypto]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- json_serializable 6.1.4 [analyzer async build build_config collection json_annotation meta path pub_semver pubspec_parse source_gen source_helper]

transitive dependencies:
- _fe_analyzer_shared 34.0.0 [meta]
- analyzer 3.2.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.2 [crypto path]
- args 2.1.0
- async 2.9.0 [collection meta]
- barcode 2.2.0 [meta qr]
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.2.1 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 3.0.1 [built_collection built_value http_multi_server logging path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.6 [analyzer async build crypto graphs logging path package_config pool pub_semver stream_transform yaml]
- build_runner_core 7.2.3 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pool timing watcher yaml]
- built_collection 5.1.1
- built_value 8.1.4 [built_collection collection fixnum meta]
- cached_network_image_platform_interface 1.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.1 [flutter flutter_cache_manager cached_network_image_platform_interface]
- characters 1.2.1
- charcode 1.3.1
- charts_common 0.12.0 [collection intl logging meta vector_math]
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- chewie 1.3.3 [cupertino_icons flutter provider video_player wakelock]
- chewie_audio 1.3.0 [cupertino_icons flutter video_player]
- cli_util 0.3.0 [meta path]
- clock 1.1.1
- cloud_firestore_platform_interface 5.4.13 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.6.8 [cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- cloud_functions_platform_interface 5.1.16 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.3.5 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- convert 3.0.0 [typed_data]
- crypto 3.0.1 [collection typed_data]
- csslib 0.17.1 [source_span]
- dart_style 2.2.1 [analyzer args path pub_semver source_span]
- fake_async 1.3.1 [clock collection]
- ffi 1.1.2
- file 6.1.0 [meta path]
- firebase_auth_platform_interface 6.1.11 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.7 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.5.1 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.7.2 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_storage_platform_interface 4.0.14 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 3.2.8 [async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- fixnum 1.0.0
- flutter_blurhash 0.7.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_layout_grid 1.0.6 [flutter collection meta quiver]
- flutter_math_fork 0.5.0 [flutter flutter_svg provider meta collection tuple]
- flutter_svg 0.23.0+1 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- frontend_server_client 2.1.2 [async path]
- glob 2.0.1 [async collection file path pedantic string_scanner]
- google_sign_in_platform_interface 2.1.1 [flutter quiver]
- google_sign_in_web 0.10.0 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- graphs 2.1.0 [collection]
- html 0.15.0 [csslib source_span]
- http_multi_server 3.2.0 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.2 [archive meta xml]
- io 1.0.3 [meta path string_scanner]
- js 0.6.4
- list_ext 1.0.2 [quiver collection]
- logging 1.0.2
- matcher 0.12.12 [stack_trace]
- material_color_utilities 0.1.5
- meta 1.8.0
- mime 1.0.1
- nested 1.0.0 [flutter]
- numerus 1.1.1
- octo_image 1.0.2 [flutter flutter_blurhash]
- package_config 2.0.0 [path]
- path 1.8.2
- path_drawing 0.5.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.2.1 [vector_math meta]
- path_provider 2.0.10 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.14 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.9 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.6 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.6 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.0
- petitparser 4.1.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.0.1 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- pub_semver 2.0.0 [collection]
- pubspec_parse 1.2.0 [checked_yaml collection json_annotation pub_semver yaml]
- qr 3.0.0 [meta]
- quiver 3.0.1 [matcher]
- riverpod 1.0.3 [collection meta state_notifier]
- shelf 1.2.0 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.2.1 [analyzer async build dart_style glob meta path source_span yaml]
- source_helper 1.3.1 [analyzer collection source_gen]
- source_span 1.9.0 [collection path term_glyph]
- sqflite 2.0.2+1 [flutter sqflite_common path]
- sqflite_common 2.2.1+1 [synchronized path meta]
- stack_trace 1.10.0 [path]
- state_notifier 0.7.0 [meta]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.1 [source_span]
- synchronized 3.0.0+2
- term_glyph 1.2.1
- test_api 0.4.12 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- time 2.1.0 [clock]
- timing 1.0.0 [json_annotation]
- tuple 2.0.0 [quiver]
- typed_data 1.3.0 [collection]
- url_launcher_android 6.0.15 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.15 [flutter url_launcher_platform_interface]
- url_launcher_linux 2.0.0 [flutter]
- url_launcher_macos 2.0.0 [flutter]
- url_launcher_platform_interface 2.0.4 [flutter plugin_platform_interface]
- url_launcher_web 2.0.0 [url_launcher_platform_interface meta flutter flutter_web_plugins]
- url_launcher_windows 2.0.0 [flutter]
- vector_math 2.1.2
- video_player 2.2.19 [flutter html video_player_android video_player_avfoundation video_player_platform_interface video_player_web]
- video_player_android 2.3.4 [flutter video_player_platform_interface]
- video_player_avfoundation 2.3.4 [flutter video_player_platform_interface]
- video_player_platform_interface 5.0.0 [flutter plugin_platform_interface]
- video_player_web 2.0.10 [flutter flutter_web_plugins video_player_platform_interface]
- wakelock 0.6.1+2 [flutter meta wakelock_macos wakelock_platform_interface wakelock_web wakelock_windows]
- wakelock_macos 0.4.0 [flutter flutter_web_plugins wakelock_platform_interface]
- wakelock_platform_interface 0.3.0 [flutter meta]
- wakelock_web 0.4.0 [flutter flutter_web_plugins js wakelock_platform_interface]
- wakelock_windows 0.2.0 [flutter wakelock_platform_interface win32]
- watcher 1.0.0 [async path pedantic]
- web_socket_channel 2.1.0 [async crypto stream_channel]
- webview_flutter 2.8.0 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]
- webview_flutter_android 2.8.8 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 1.8.0 [flutter plugin_platform_interface]
- webview_flutter_wkwebview 2.7.5 [flutter path webview_flutter_platform_interface]
- win32 2.6.1 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 5.1.1 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]```

</details>

---

**Comment by darshankawar** [comment]
Thanks for the report @hparicha 
I tried to replicate it but didn't get the error you reported. Below are the steps I took:

1. Used `cloud_firestore` plugin example.
2. Added `cloud_functions` latest version plugin to it.
3. Added `FirebaseFunctions.instance.useFunctionsEmulator('localhost', 5001);` in `main()` method.
4. Setup and enabled Functions emulator using `firebase init` -> `firebase emulators: start`as below:

<img width="569" alt="Screenshot 2022-09-12 at 1 57 35 PM" src="https://user-images.githubusercontent.com/67046386/189608350-3328d04a-c8ca-4ed5-882d-efb3d42b4abb.png">
<img width="405" alt="Screenshot 2022-09-12 at 1 58 19 PM" src="https://user-images.githubusercontent.com/67046386/189608358-f880402c-3239-4129-ac77-4c577301a42a.png">

5. With emulator running, ran `flutter run -d chrome` which compiled and built the code in step # 1 properly without any issues:

```
dhs@Dhss-MacBook-Pro cloud_firestore_example % flutter run -d chrome          
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             20.6s
This app is linked to the debug service: ws://127.0.0.1:50720/vq5c-YZXVPY=/ws
Debug service listening on ws://127.0.0.1:50720/vq5c-YZXVPY=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:50720/vq5c-YZXVPY=
Flutter Web Bootstrap: Auto
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:50720/vq5c-YZXVPY=

```

<img width="629" alt="Screenshot 2022-09-12 at 1 56 59 PM" src="https://user-images.githubusercontent.com/67046386/189608576-074d6f5c-8f0b-4965-aece-32847279bfe8.png">

I am on latest master version of the framework.

If above steps don't reflect the issue in your case, please provide detailed steps to replicate so that we can debug it further.

**Comment by hparicha** [comment]
Thank you for looking @darshankawar 
I upgraded my outdated packages (and then did `flutter pub upgrade --major-versions`).
My web app runs again now so can close this off.
